src/org/loklak/api/search/TweetCheckServlet.java

```diff
@@ -0,0 +1,66 @@
+package org.loklak.api.search;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import org.json.JSONObject;
+import org.loklak.data.DAO;
+import org.loklak.harvester.Post;
+import org.loklak.http.RemoteAccess;
+import org.loklak.server.Query;
+
+
+public class TweetCheckServlet extends HttpServlet{
+    private static final long serialVersionUID = 4568123491056781234L;
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        Query query = RemoteAccess.evaluate(request);
+
+        // manage DoS
+        if (query.isDoS_blackout()) {response.sendError(503, "your request frequency is too high"); return;}
+
+        String callback = query.get("callback", "");
+        boolean jsonp = callback != null && callback.length() > 0;
+        boolean minified = query.get("minified", false);
+
+        String tweetQuery = query.get("tweet", "");
```

Comment:
Does the user put the entire link here? or just the `tweetid`? If it's the entire link, why is it designed this way? In what scenarios will someone be using this API call with a `full` URL string.

In ideal cases if you were looking up something, you'd look things up by ID and not by the entire path to the ID right, the same should be the analogy here. Making this just the `tweetID:string` will make `parseTweetId()` a redundant method.

Reply:
The issue said: 
- `/api/show.json?tweet=https://twitter.com/ArtisticBondage/status/885041783079661569`
- `/api/show.json?tweet=885041783079661569`

Now if it's a link, we extract the id, and if it's already id, we are good to go.

So I added this method

Reply:
The question here would be how would you know from the current method that the `tweet` value is a valid ID or a link. Doing a check for `http` contained in the string would not help here at all. I can pass my string to be `tweet=https://126312356123612/ajsdba/1 which will pass and it'll treat the ID as `1`  but this has major problems because the URL isn't by itself a valid URL.

You need to parse `https://twitter.com/{:user}/status/{:id}` for validation.